ConfigSpace/forbidden.py

```diff
@@ -72,12 +74,13 @@ def is_forbidden(self, instantiated_hyperparameters):
 
 
 class AbstractForbiddenClause(AbstractForbiddenComponent):
-    def get_descendant_literal_clauses(self):
+    def get_descendant_literal_clauses(self) -> List['AbstractForbiddenClause']:
```

Comment:
Could you please check with the debugger and the unittests whether this actually returns a list of `AbstractForbiddenClause`? I would assume that the list can contain `AbstractForbiddenComponent` as well.